build,win: fix Python detection on localized OS

joaocgreis opened the pull request:
This code previously assumed the default value was always printed to the console by reg.exe as `(default)`, but this is not true on localized versions of Windows and can contain spaces.

Fixes: https://github.com/nodejs/node/issues/29417

cc @targos @nodejs/platform-windows 

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

targos commented:
CI failed on Windows:
```
18:05:18 > "C:\Python27\\python.exe" -V   2>&1  | findstr /R "^Python.2.*"  1>NUL 
18:05:18 
18:05:18 > EXIT /B 0 
18:05:18 The system cannot find the batch label specified - found-python2
18:05:18 The system cannot find the batch label specified - done-path-v2
18:05:18 
18:05:18 > if ERRORLEVEL 1 exit /b
```

joaocgreis commented:
This was an issue with the CI machine that was used to compile. The git option `core.autocrlf` should be set to `true` as is default on Windows, but on that machine was `false`. I've fixed the machine and added a fixup to enforce that option for `find_python.cmd`, similar to what already happens for `vcbuild.bat`. The problem running the `cmd` file was that `goto` does not work well when the file has Unix line endings.

Trott commented:
Landed in 7bff962
